src/Arcus.WebApi.Logging/RequestTrackingMiddleware.cs

```diff
@@ -134,11 +136,22 @@ private static async Task<string> SanitizeRequestBodyAsync(Stream requestStream)
                 }
             }
 
-            using (var reader = new StreamReader(requestStream))
+            var reader = new StreamReader(requestStream);
+            string contents = await reader.ReadToEndAsync();
+
+            try
             {
-                string contents = await reader.ReadToEndAsync();
-                return contents;
+                if (requestStream.CanSeek)
+                {
+                    requestStream.Seek(originalPosition, SeekOrigin.Begin);
+                }
             }
+            catch
+            {
+                // Nothing to do here, we want to ensure the value is always returned.
```

Comment:
What exception would you expect here ?  And why not log the exception should one occur ?

Reply:
For me this is best effort. The app should not return 500 nor log exception (= alert) on this.

But I could live with a warning if you prefer.

Reply:
Indeed, you don't want to have this logged for every request.
But then I was thinking; why not add a guard clause at the beginning of the method which verifies if the stream is seekable.  If not, return there already.   If the string is not seekable, you're going to read the request stream here, and the stream will be positioned at the end, making it unusable for later processing.

```
private static async Task<string> SanitizeRequestBodyAsync(Stream requestStream)
{	            
    if (!requestStream.CanRead)	            
    {	   
        return null;	           
    }	            

    if( !requestStream.CanSeek )
    {
        return null;
     }
```

Reply:
What if it's in the middle of the stream (for some reason), do you still want to not look at it?

Reply:
If the stream is not seekable, you're not able to reposition it to it's original position.   Do you want to read the stream then anyway and possibly cause side-effects ?

Reply:
Good point, might align with you. Thoughts @stijnmoreels ?

Reply:
Yes, I think @fgheysels has a point. Shortcutting when not seekable and logging this?

Reply:
I've changed it but used warnings instead so that it's more explicit and an empty body is tracked.

Thoughts about this? Reason is that otherwise you will only notice if you check the tracked request and not have "indicators" of issues.

Reply:
I'm just thinking on how I'd use it myself:

Suppose I have an API call that fails for some reason or another, I'd inspect the logs and one of the things that I might want to check is the request body.    If the body has been logged: good.  If it hasn't, and instead of the body being present in this property a string is present which says that the request body could not be logged because of this or that reason:  too bad, but at least I know why it hasn't been logged without having to inspect any other logs to find out why the body is not present in the logs.

I don't think that having this kind of information as a separate warning gives any additional value.  I'm also not going to search actively in my logs for requests for which the request body could not be logged I think :).

Reply:
Makes sense!